samplers/jaegerremote/sampler_remote_options.go

```diff
@@ -49,15 +49,13 @@ func newConfig(options ...Option) config {
 			OperationNameLateBinding: defaultSamplingOperationNameLateBinding,
 		},
 	}
+	c.updaters = append([]samplerUpdater{&perOperationSamplerUpdater{
+		MaxOperations:            c.posParams.MaxOperations,
+		OperationNameLateBinding: c.posParams.OperationNameLateBinding,
+	}}, c.updaters...)
```

Comment:
This shouldn't be moved before the options loop. Doing so means any updater specified in `WithUpdaters` will never be used.

Reply:
> This shouldn't be moved before the options loop. Doing so means any updater specified in `WithUpdaters` will never be used.

`withUpdaters` will overwrite `updaters`, so it has the highest priority. On the contrary, putting it after options will change the effect of `withUpdaters`, which will fail some tests.

Reply:
`withUpdaters` is not exported and only used in tests.  Fix the tests that use it.  This absolutely needs to happen after the options are applied as `WithMaxOperations` and `WithOperationNameLateBinding` _are_ exported and would not be effective if this is done before option application.

Reply:
got it, done, thanks for your explanation.